taskvine/src/bindings/python3/ndcctools/taskvine/dask_dag.py

```diff
@@ -102,17 +102,33 @@ def depth_of(self, key):
     def initialize_graph(self):
         for key, sexpr in self._working_graph.items():
             self.set_relations(key, sexpr)
+        for key, sexpr in self._working_graph.items():
+            self.set_depth(key)
 
-    def find_dependencies(self, sexpr, depth=0):
+    def find_dependencies(self, sexpr):
         dependencies = set()
         if self.graph_keyp(sexpr):
             dependencies.add(sexpr)
-            self._depth_of[sexpr] = min(depth, self._depth_of[sexpr])
         elif not DaskVineDag.symbolp(sexpr):
             for sub in sexpr:
-                dependencies.update(self.find_dependencies(sub, depth + 1))
+                dependencies.update(self.find_dependencies(sub))
         return dependencies
 
+    def set_depth(self, key):
```

Comment:
I'm not sure about this change. Didn't the old way of computing depth worked?

Reply:
@btovar No, the old way didn't give me the right depth.

Reply:
It did work for me when I was testing checkpoints. Please check again because it may indicate that something else in the code broke. Also, it is a much easier code to follow, which may help with dealing with future bugs.